### PR TITLE
Add a priority queue allowlist check. URL pointing to allowlist is...

### PR DESCRIPTION
**Pull request by ben-alkov**
...read from a config file located on the koji hub,
'/etc/koji-hub/plugins/koji-containerbuild.conf', which looks like

```config
[priority_allowlist]
priority_target_list_url = https://foo/bar.yaml
```

Allowlist looks like

```yaml
targets:
- managed-services-containers
- ryan-is-special
```

Signed-off-by: Ben Alkov <ben.alkov@redhat.com>

- CLOUDBLD-3573

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

**Comment by ben-alkov** [comment]
@lcarva, @rcerven; WDYT so far? No ACKs, please, just feedback.

File: koji_containerbuild/plugins/hub_containerbuild.py [hunk 1]
```diff
@@ -53,6 +58,30 @@ def _get_task_opts_and_opts(opts, priority, channel):
...
+    priority_target_list = CONFIG.get('priority_allowlist', 'priority_target_list_url')
+    req = Request(priority_target_list)
+    try:
+        response = urlopen(req)
+        prio_target_list = yaml.safe_load(response.read().decode())
+        if target in prio_target_list['targets']:
+            if not priority:
+                priority = koji.PRIO_DEFAULT - 1
+            else:
+                priority -= 1
```
Comment (MartinBasti) [inline koji_containerbuild/plugins/hub_containerbuild.py:73]: What happen when priority is 0 or -1 ?

File: koji_containerbuild/plugins/hub_containerbuild.py [hunk 2]
```diff
@@ -53,6 +58,30 @@ def _get_task_opts_and_opts(opts, priority, channel):
 
     return opts, taskOpts
 
+
+def _check_priority_allowlist(target, priority):
+    CONFIG = koji.read_config_files([(CONFIG_FILE, True)])
+    priority_target_list = CONFIG.get('priority_allowlist', 'priority_target_list_url')
+    req = Request(priority_target_list)
+    try:
+        response = urlopen(req)
```
Comment (MartinBasti) [inline koji_containerbuild/plugins/hub_containerbuild.py:67]: Is there any read time limit? I.e not waiting 2 minutes for TCP timeout, etc...

File: koji_containerbuild/plugins/hub_containerbuild.py (hunk 1 above)
Comment (ben-alkov) [reply koji_containerbuild/plugins/hub_containerbuild.py:73]: AIUI, both are valid; "only admins have the right to specify a negative priority"

File: koji_containerbuild/plugins/hub_containerbuild.py (hunk 2 above)
Comment (ben-alkov) [reply koji_containerbuild/plugins/hub_containerbuild.py:67]: I'll check

Comment (ben-alkov) [reply koji_containerbuild/plugins/hub_containerbuild.py:67]: Apparently only for Py 3. Another option is switching to `requests`.

Comment (ben-alkov) [reply koji_containerbuild/plugins/hub_containerbuild.py:73]: Fixed using `socket`'s global timeout. Works for 2 & 3.

File: koji_containerbuild/plugins/hub_containerbuild.py [hunk 3]
```diff
@@ -53,6 +64,33 @@ def _get_task_opts_and_opts(opts, priority, channel):
...
+        prio_target_list = yaml.safe_load(response.read().decode())
+        if target in prio_target_list['targets']:
+            if not priority:
+                priority = koji.PRIO_DEFAULT - 1
+            else:
+                priority -= 1
+    except Exception as exc:
+        logger.warning('Failed to fetch priority target list: "%s":', priority_target_list)
+        if hasattr(exc, 'reason'):
+            if isinstance(exc.reason, socket.timeout):
```
Comment (MartinBasti) [inline koji_containerbuild/plugins/hub_containerbuild.py:83]: is this really `isinstance(exc.reason ...)` and not just `isinstance(exc, ...)`?

Comment (ben-alkov) [reply koji_containerbuild/plugins/hub_containerbuild.py:83]: Truly it is. Tested with `timeout = 0.01` in Jupyter.

**Comment by ben-alkov** [comment]
Closing as Dropped.

**Pull request closed without merging**